### PR TITLE
Added User Reference support

### PR DESCRIPTION
**Pull request by vishalshrm539**
* Added User Reference support.
* Please refer US-491194-1 for more details.

### CODE REVIEW

File: src/components/forms/UserReference/index.ts [hunk 1]
```diff
@@ -1,186 +1,273 @@
...
-      }`;
+      PCore.getRestClient()
+        .invokeRestApi("getListData", { queryPayload })
+        .then((resp) => {
+          const ddDataSource = resp.data.data.map((listItem) => ({
+            key: listItem.pyUserIdentifier,
+            value: listItem.pyUserName,
+          }));
+          this.dropDownDataSource = ddDataSource;
+          this.dropDownDataSource = JSON.stringify(this.dropDownDataSource);
```
Comment (tumms2021389) [inline src/components/forms/UserReference/index.ts:196]: why are you changing it as string?

File: src/components/forms/Dropdown/index.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ declare var PCore: any;
 class Dropdown extends FormComponentBase {
 
   @property( {attribute: false, type: Array} ) options;
+  @property( {attribute: true, type: Array} ) datasource = [];
```
Comment (tumms2021389) [inline src/components/forms/Dropdown/index.ts:21]: you are passing string value to the prop I guess

File: src/components/forms/UserReference/index.ts (hunk 1 above)
Comment (vishalshrm539) [reply src/components/forms/UserReference/index.ts:196]: I was passing the array before but that didn't work then I'd to convert it to a string first in parent then parse it again in the child component.

File: src/components/forms/Dropdown/index.ts (hunk 2 above)
Comment (vishalshrm539) [reply src/components/forms/Dropdown/index.ts:21]: Yes, thanks for noticing this, will update.

**Review by JEAxero — APPROVED** [review]
SDK-WC UserReference passes now

**Pull request merged**